posthog/src/main/java/com/posthog/java/QueueManager.java

```diff
@@ -0,0 +1,92 @@
+package com.posthog.java;
+
+import java.time.Instant;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+
+import org.json.JSONObject;
+
+public class QueueManager implements Runnable {
+    class QueuePtr {
+        // TODO: not sure if there's a datastructure in Java that gives me these
+        // operations efficiently, specifically retrieveAndReset
+        // Not sure if LinkedList and JSONObject are good choices here
+        public List<JSONObject> ptr = new LinkedList<JSONObject>();
+
+        public QueuePtr() {
+        };
+
+        public synchronized int size() {
+            return ptr.size();
+        }
+
+        public synchronized boolean isEmpty() {
+            return ptr.isEmpty();
+        }
+
+        public synchronized void add(JSONObject json) {
+            ptr.add(json);
+        }
+
+        public synchronized List<JSONObject> retrieveAndReset() {
+            if (isEmpty()) {
+                return Collections.emptyList();
+            }
+            List<JSONObject> cur = ptr;
+            ptr = new LinkedList<JSONObject>();
```

Comment:
Why are we mixing `LinkedList` with `List`? 

(side note: I haven't written Java in a while)

Reply:
`List` is an interface (https://docs.oracle.com/javase/8/docs/api/java/util/List.html) so we can't create an object of `List` directly, but we have to pick one of the implementing classes, e.g. `LinkedList` to create an instance, however I do want this class to externally visibly operate on the `List` interface, so I can change the implementation without impacting any users, e.g. say I later want to change the `LinkedList` to  `ArrayList` I only need to make changes within this class.